tests/test_interpolation.py

```diff
@@ -347,11 +348,106 @@ def test_incremental_dict_with_interpolation() -> None:
         ({"list": ["${ref}"], "ref": "bar"}, "list.0", "bar"),
     ],
 )
-def test_interpolations(cfg: DictConfig, key: str, expected: Any) -> None:
+def test_interpolations(cfg: Dict[str, Any], key: str, expected: Any) -> None:
     c = OmegaConf.create(cfg)
     assert OmegaConf.select(c, key) == expected
 
 
+@pytest.mark.parametrize(  # type: ignore
+    "cfg,expected_dict",
+    [
+        pytest.param(
+            """
+            a: 1
+            b: a
+            c: ${${b}}
+            """,
+            {"c": 1},
+            id="basic nesting",
+        ),
+        pytest.param(
+            """
+            a: OMEGACONF
+            b: NESTED_INTERPOLATIONS_TEST
+            c: ${env:${a}_${b}}
+            """,
+            {"c": "test123"},
+            id="nesting with key",
+        ),
+        pytest.param(
+            """
+            x: 1
+            y: 2
+            op: plus
+            z: ${plus:${x},${y}}
+            t: ${${op}:${x},${y}}
+            """,
+            {"z": 3, "t": 3},
+            id="nesting with (possibly dynamic) resolver",
+        ),
+        pytest.param(
+            """
+            a:
+                b: 1
+                c: 2
+                d: ${a.b}
+            b: c
+            c: ${a.${b}}
+            d: ${${b}}
+            e: .d
+            f: ${a${e}}
+            """,
+            {"c": 2, "d": 2, "f": 1},
+            id="member access",
+        ),
+        pytest.param(
+            """
+            a: def
+            b: abc_{${a}}
+            """,
+            {"b": "abc_{def}"},
+            id="braces in string",
+        ),
+        pytest.param(
+            """
+            a: A
+            b: ${env:x=A}
+            c: ${env:x=${a}}
+            """,
+            # This behavior may be a bit surprising but it is how it works now. If
+            # it changes in the future we should ensure that both `b` and `c` yield
+            # the same result.
+            {"b": "${env:x=A}", "c": "${env:x=A}"},
+            id="illegal character in interpolation",
+        ),
+    ],
+)
+def test_nested_interpolations(cfg: str, expected_dict: Dict[str, Any]) -> None:
+    os.environ["OMEGACONF_NESTED_INTERPOLATIONS_TEST"] = "test123"
+    OmegaConf.register_resolver("plus", lambda x, y: int(x) + int(y))
+    try:
+        c = OmegaConf.create(cfg)
+        for key, expected in expected_dict.items():
+            assert OmegaConf.select(c, key) == expected
+    finally:
+        OmegaConf.clear_resolvers()
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "cfg,key",
+    [
+        # All these examples have non-matching braces.
+        ({"a": "PATH", "b": "${env:${a}"}, "b"),
```

Comment:
can also add pytest.param with ids here.

Reply:
Done in f25f17f